examples.md

```diff
@@ -99,6 +100,28 @@ We cache the elements of the Cabal store separately, as the entirety of `~/.caba
     key: ${{ runner.os }}-${{ matrix.ghc }}-dist-newstyle
 ```
 
+## Haskell - Stack
+
+```yaml
+- uses: actions/cache@v1
+  name: Cache ~/.stack
+  with:
+    path: ~/.stack
+    key: ${{ runner.os }}-homestack-${{ hashFiles('**/stack.yaml.lock') }}-${{ hashFiles('**/package.yaml') }}
+    restore-keys: |
+      ${{ runner.os }}-homestack-${{ hashFiles('**/stack.yaml.lock') }}-
+      ${{ runner.os }}-homestack-
+- uses: actions/cache@v1
+  name: Cache .stack-work
+  with:
+    path: .stack-work
+    key: ${{ runner.os }}-stackwork-${{ hashFiles('**/stack.yaml.lock') }}-${{ hashFiles('**/package.yaml') }}-${{ hashFiles('**/*.hs') }}
```

Comment:
Hashing over the source file will create a new cache key for each change. It'll work with restore keys, but the [maximum size limit of 5 GB for all github caches in the repository](https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy) might be an issue. Would you advice to remove a key containing a hash over source files?